include/fc/io/cfile.hpp

```diff
@@ -27,14 +27,21 @@ class cfile_datastream;
 class cfile {
 public:
    cfile()
-     : _file(nullptr, &fclose)
+     : _open(false),
+       _file_path(""),
+       _file(nullptr, &fclose)
    {}
 
+   ~cfile() {
+       if(_open)
+           fclose(_file.get());
+   }
+
```

Comment:
indentation seems to be 4 spaces instead of 3 spaces as used throughout rest of code. Can you please fix.